### ISSUE TITLE
Default Query - Should be Submitting with "retrieveFacets = true"

### ISSUE THREAD

**Issue by rosiegrant** [yext/answers-search-ui#369]
If we submit a default query and the page has a facets component, the `retrieveFacets` parameter on the API call should always be set to `true` 

Otherwise the default queries that we submit won't show facets:
https://answers-js-demo.netlify.com/sites/rose/employees.html
![image](https://user-images.githubusercontent.com/22160788/69088799-921a2780-0a15-11ea-8c52-d76a5ff0b5e5.png)

**Comment by data-enabler** [comment]
Looks like `retrieveFacets` doesn't get enabled until a Facets component gets added to the page, meaning that there's an ordering dependency here 😞. Haven't confirmed, but I suspect it would work if the Facets component were added before the SearchBar.

**Comment by data-enabler** [comment]
Noticed that it stopped happening when I tried to reproduce it in 0.10.1, and it's because the location permission check added in #387 causes the initial search to happen asynchronously (and therefore after both components have been added to the page).

There are plans to add a `retrieveFacets` option to the top-level search config, but in the meantime I think making initial search always asynchronous (currently it isn't in browsers that don't support the permissions API) should fix this for 99% of cases (and make the behavior more consistent). Basically, as long as the Facets component isn't added asynchronously after the SearchBar, it should work as expected.
